### PR TITLE
Fix #865 Re-org the source tree

### LINKED ISSUE #865: Re-org the source tree
## Goals

* Reduce the number of extensions under `extensions`
* Increase the visibility of the extensions now under `extensions/support`. They are hard to find for newcomers and they are hard to scroll to for maintainers

## Non-goals (possible future goals)

* Split the extensions under `extensions` into even smaller groups, like we have in the GH workflow
* Do anything with the itests

## Current state:

```
├── catalog
├── docs
├── examples
├── extensions
│   ├── ...
│   ├── support
├── integration-tests
├── poms
└── tooling
```

## Proposed state

```
├── catalog
├── docs
├── examples
├── extensions
├── extensions-core       # The extensions whose camel artifacts are under core in the Camel source tree
├── extensions-support   # Moved from extensions/support
├── integration-tests
├── poms
└── tooling

```

WDYT?

### PR DESCRIPTION
**Pull request by ppalaga**
Fix #865

### CODE REVIEW

File: docs/modules/ROOT/pages/list-of-camel-quarkus-extensions.adoc [hunk 1]
```diff
@@ -312,49 +312,31 @@ Number of Camel data formats: 23 in 18 JAR artifacts (0 deprecated)
 == Camel Languages
 
 // languages: START
-Number of Camel languages: 11 in 5 JAR artifacts (0 deprecated)
+Number of Camel languages: 3 in 3 JAR artifacts (0 deprecated)
```
Comment (jamesnetherton) [inline docs/modules/ROOT/pages/list-of-camel-quarkus-extensions.adoc]: Is it expected that extensions are removed from here? We should be listing all of the stuff that camel-quarkus supports.

Comment (ppalaga) [reply docs/modules/ROOT/pages/list-of-camel-quarkus-extensions.adoc]: No, thanks for spotting it!

**Comment by ppalaga** [comment]
202430f eliminates the missing languages in the .adoc files.

File: tooling/scripts/sort-poms.groovy [hunk 2]
```diff
@@ -187,14 +188,16 @@ void sortModules(String[] sortModulesPaths) {
 
 void updateMvndRules(String[] updateMvndRuleDirs) {
     final Set<String> extensionArtifactIds = [] as TreeSet
-    Files.list(baseDir.resolve('extensions'))
+    Stream.of('extensions', 'extensions-core')
```
Comment (lburgazzoli) [inline tooling/scripts/sort-poms.groovy]: not very groovysh :)

Comment (ppalaga) [reply tooling/scripts/sort-poms.groovy]: Yes, I am a proper Groovy noob! If there was a plugin to run Java as a script, I'd very much prefer writing it in Java :)

**Pull request merged**